Sources/MapboxCoreNavigation/Directions+RoutingProvider.swift

```diff
@@ -31,8 +31,24 @@ extension Directions: RoutingProvider {
     @discardableResult public func calculateRoutes(options: MatchOptions, completionHandler: @escaping MatchCompletionHandler) -> NavigationProviderRequest? {
         return calculate(options, completionHandler: completionHandler)
     }
-    
+
     @discardableResult public func refreshRoute(indexedRouteResponse: IndexedRouteResponse, fromLegAtIndex: UInt32, completionHandler: @escaping RouteCompletionHandler) -> NavigationProviderRequest? {
+        _refreshRoute(indexedRouteResponse: indexedRouteResponse,
+                      fromLegAtIndex: Int(fromLegAtIndex),
+                      currentRouteShapeIndex: nil,
+                      currentLegShapeIndex: nil,
+                      completionHandler: completionHandler)
+    }
+
+    @discardableResult public func refreshRoute(indexedRouteResponse: IndexedRouteResponse, fromLegAtIndex: UInt32, currentRouteShapeIndex: Int, currentLegShapeIndex: Int, completionHandler: @escaping RouteCompletionHandler) -> NavigationProviderRequest? {
```

Comment:
Tracking the route shape index and leg shape index separately might be prone to error. Could this method take just a route shape index and compute the leg shape index based on `RouteLeg.segmentRangesByStep`? Unfortunately, MapboxDirections doesn’t provide a `Route.segmentRangesByLeg`, but we could add one to this module for now.

Reply:
This data is calculated in NN, delivered via NavigatorStatus, and propagated further with RouteProgress. It's coupled along the way and recalculating it could introduce another set of errors.